### PR TITLE
ui-core: add build instructions to readme file

### PR DESCRIPTION
**Pull request by ghost**
Issue #89

### CODE REVIEW

**Comment by ghost** [comment]
I struggled a lot today to build the project. There may be other commands that are better. Is there something that should be added or removed?

Screenshot 2024-07-06 at 8 46 11 PM

I could not resolve the problems below with this library. What am I doing something wrong? I tried to install the package as specified on the npm page, but I had no luck: https://www.npmjs.com/package/@osrd-project/ui-icons

Screenshot 2024-07-06 at 8 46 59 PM

**Comment by ghost** [comment]
@sim51 @emersion I seem not able to request a review. Would you please review this pr? :)

File: README.md [hunk 1]
```diff
@@ -2,6 +2,20 @@
...
+# Build Instructions
+
+1. Go to the project's `root` directory in your terminal
+2. Run `sudo chown -R $(whoami) ~/.npm`
+3. Enter your computer's password
+4. Run `rm -rf node_modules package-lock.json`
+5. Run `npm cache clean --force`
+6. Run `npm install @nx/nx-darwin-x64`
+7. Run `npm install @rollup/rollup-darwin-x64`
+8. Run `npm run clean`
```
Comment (emersion) [inline README.md]: Hm, I don't think any of these should be necessary...

- `.npm` should never have files owned by root. `npm install -g` may write files to this directory but shouldn't be run as root.
- `package-lock.json` is important to keep to install exactly the right dependencies and don't implicitly update everything.
- Cleaning the cache shouldn't be required unless the cache has been previously corrupted somehow by a previous command (that's outside the scope of a normal build).
- Not sure why Darwin-specific dependencies are explicitly pulled in, they should get obtained implicitly via `npm install` (without arguments).

Does that make sense?

File: README.md [hunk 2]
```diff
@@ -2,6 +2,20 @@
...
+9. Run `npm run build`
+10. Run `npm run test`
```
Comment (emersion) [inline README.md]: Running the tests isn't strictly required to build the project. Maybe could be mentioned in another sentence?

File: README.md [hunk 3]
```diff
@@ -2,6 +2,20 @@
 
 Icons, fonts, colors, components and all user interface items for OSRD project.
 
+# Build Instructions
+
+1. Go to the project's `root` directory in your terminal
```
Comment (emersion) [inline README.md]: I personally find this sentence confusing: I'd try to find a directory named "root", but there is none.

In general I think it's fine to assume that the user is initially already in the project's directory.

**Comment by emersion** [comment]
It seems like this PR touches more than just the README - it also includes some `package.json` and `package-lock.json` changes. As said above, I don't think we need these changes.

To sum things up, IMHO `npm install` + `npm run build` + `npm run storybook` should be enough to get people started.

File: README.md (hunk 1 above)
Comment (ghost) [reply README.md]: Yes, Emersion. That all makes sense. Thanks for the feedback. I appreciate you taking the time. 

My local linter, I think, is preventing me from not formatting the existing README file, but I only added the instructions you recommended.

File: README.md (hunk 2 above)
Comment (ghost) [reply README.md]: Good point. Thanks :)

File: README.md (hunk 3 above)
Comment (ghost) [reply README.md]: Yup. Agreed. Fixed.

**Comment by Yohh** [comment]
@turnerian2004 that's a good point, basically we run the project with `npm run start --ws` (after `npm install` and `npm run build`) in order to load every workspace in the project, that's not an issue if you need to work for only one specific sub-project, but some of them depends on other ones (`osrd-manchette` needs `osrd-spacetimechart`, `osrd-speedspacechart` needs `osrd-core`, many of them needs `osrd-icons`...)

many changes are work-in-progress by @jacomyal , and a different process will be set very soon, but I think it's a good start!

File: README.md [hunk 4]
```diff
@@ -8,37 +8,62 @@ Development will leverage preconstruct. **Documentation yet to come**.
...
 
-If this section is not added, the package will not be published. Having a `"private": true` in the package.json will also prevent the package from being published.
+If this section is not added, the package will not be published. Having a `"private": true` in the
+package.json will also prevent the package from being published.
+
+# Build Instructions
+
+1. Run `npm install`
+2. Run `npm run build`
+3. Run `npm run storybook`
```
Comment (Yohh) [inline README.md:65]:
```suggestion

# Start a specific project

1. Ryn `cd ui-<name-of-the-project>`
2. Run `npm run storybook`

# start every pprojects

1. Run `npm run start --ws`
```

Comment (Yohh) [inline README.md:65]:
```suggestion
3. Run `npm run storybook`
4. Run `cd ui-<name-of-the-project>`
5. Run `npm run watch`
```

Comment (clarani) [reply README.md:65]: I think you need to run `npm run watch` only if you're developping on a repo and modifying its css right ? If it's the case, maybe we can separate in an other section

```suggestion
3. Run `npm run storybook`

While developping on a project, you need to run the following commands to see the css modifications:
4. Run `cd ui-<name-of-the-project>`
5. Run `npm run watch`
```

Comment (Yohh) [reply README.md:65]: exactly, seems better this way

**Comment by ghost** [comment]
I'll get back later this week. I want to do a little reading up, if that's okay.

**Comment by ghost** [comment]
@Yohh @emersion @clarani 

Thank you for the feedback on the pr. Much appreciated :) Feel free to make any desired changes to merge the pr :)

File: README.md [hunk 5]
```diff
@@ -8,37 +8,67 @@ Development will leverage preconstruct. **Documentation yet to come**.
...
+
+While developping on a project, you need to run the following commands to see the css modifications:
+
+1. Run `cd ui-<name-of-the-project>`
+2. Run `npm run watch`
+
+# Tests
+
+1. Run `npm run test`
```
Comment (Yohh) [inline README.md:74]: @emersion should we add this too?

```suggestion
1. Run `npm run test`

## License

OSRD and OSRD-UI are licensed under the GNU Lesser General Public License v3.0, see LICENSE.

Copyright © 2022 The OSRD Contributors
```

Comment (clarani) [reply README.md:74]: I think it is out of scope of this ticket, we may add this later if it's fine for you @Yohh :)

**Review by clarani — APPROVED** [review]
LGTM, thanks !! :white_check_mark: 

This README really needed this improvement :pray:

**Review by Yohh — APPROVED** [review]

**Pull request merged**